src/index.js

```diff
@@ -16,98 +16,54 @@ let ElasticsearchProvider = (
       not: 'must_not',
     }[group.join || 'and']
 
-    let result = {
+    return {
       bool: {
         [join]: filters,
+        ...(join === 'should' && { minimum_should_match: 1 }),
       },
     }
-    if (join === 'should') result.bool.minimum_should_match = 1
-
-    return result
   },
-  runSearch(options = {}, context, schema, filters, aggs) {
-    let query = filters
+  async runSearch({ requestorContext } = {}, node, schema, filters, aggs) {
+    let { scroll, scrollId } = node
+    let request = scrollId
+      ? // If we have scrollId then keep scrolling, no query needed
+        { scroll: scroll === true ? '2m' : scroll, scrollId }
+      : // Deterministic ordering of JSON keys for request cache optimization
+        stableKeys({
+          index: schema.elasticsearch.index,
+          // Scroll support (used for bulk export)
+          ...(scroll && { scroll: scroll === true ? '2m' : scroll }),
+          body: {
+            // Wrap in constant_score when not sorting by score to avoid wasting time on relevance scoring
+            query:
+              filters && !_.has('sort._score', aggs)
+                ? constantScore(filters)
+                : filters,
+            // If there are aggs, skip search results
+            ...(aggs.aggs && { size: 0 }),
+            // Sorting by _doc is more efficient for scrolling since it won't waste time on any sorting
+            ...(scroll && { sort: ['_doc'] }),
+            ...aggs,
+          },
+        })
 
-    // Wrapping any query NOT sorted by _score in a constant_score,
-    // so that it returns a constant score equal to the query boost
-    // for every document in the filter.
-    // Filter clauses are executed in filter context, meaning that scoring
-    // is ignored and clauses are considered for caching.
-    if (query && !_.has('sort._score', aggs))
-      query = {
-        constant_score: {
-          filter: query,
-        },
-      }
+    let requestOptions = _.merge({ headers: requestorContext }, config.request)
```

Comment:
Since this is a breaking change, now is a good time to reconsider `config.request` (set at provider instantiation, ideally once per app) and `options.requestorContext` (passed in at search time, generally once per search request). We probably don't need both, and `requestorContext` might be better named as something else since it's just the headers (e.g. `headers` 😄 )